flink-formats/flink-parquet/src/main/java/org/apache/flink/formats/parquet/vector/reader/TimestampColumnReader.java

```diff
@@ -109,6 +157,37 @@ public static TimestampData int96ToTimestamp(
         }
     }
 
+    public static TimestampData int64ToTimestamp(
+            boolean utcTimestamp, long value, LogicalTypeAnnotation.TimeUnit timeUnit) {
+        long nanosOfMillisecond = 0L;
+        long milliseconds = 0L;
+
+        switch (timeUnit) {
+            case MILLIS:
+                milliseconds = value;
+                nanosOfMillisecond = value % MILLIS_PER_SECOND * NANOS_PER_MILLISECOND;
+                break;
+            case MICROS:
+                milliseconds = value / MICROS_PER_MILLISECOND;
+                nanosOfMillisecond = (value % MICROS_PER_SECOND) * NANOS_PER_MICROSECONDS;
+                break;
+            case NANOS:
+                milliseconds = value / NANOS_PER_MILLISECOND;
+                nanosOfMillisecond = value % NANOS_PER_SECOND;
+                break;
+            default:
+                break;
```

Comment:
No time unit means drop the value? Should we throw an error?

Reply:
Good catch. LogicalTypeAnnotation.TimeUnit has only the 3 values that are all covered, so there should neither be a practical danger of dropping the value nor big risk of throwing an exception under default for clarity. It is also more of a stylistic concern that we can deal with in the master branch. I will open a PR there.

Reply:
I would throw an exception for defensive programming.